thin-replica-server/include/thin-replica-server/thin_replica_impl.hpp

```diff
@@ -120,6 +120,9 @@ class ThinReplicaImpl {
   // last timestamp when subscription status for live updates was not ok
   std::optional<std::chrono::steady_clock::time_point> last_failed_subscribe_status_time;
 
+  // TBD Persist to resume
+  uint64_t last_sent_block_id_{0};
```

Comment:
We will need to keep a map of this metric for a client ID and stream_type. How would we know otherwise if we are correctly updating the metric? 
Client1 subscribed, last sent block ID was 100. Client2 subscribed to block 0, hasn't received any blocks, we are initializing the metric to 100 incorrectly!

Reply:
I am not comparing b/w live updates and storage updates. My main concern is at TRS serves multiple clients at a time, and when we look at `last_sent_block_id` metric, we should know for which client and for which stream `last_sent_block_id` corresponds to, when it is initially set [here](https://github.com/vmware/concord-bft/blob/3b66c30c521980f85f39058c2cdbb64a960b50d2/thin-replica-server/include/thin-replica-server/thin_replica_impl.hpp#L294) on a new subscription. Otherwise, we will be incorrectly setting the metric if the same TRS serves multiple clients and/or multiple streams.

> Tags will help us conclude that TRS served client 1 up to block id 100 and then it started serving history updates to client 2 starting from 0.

if the private variable `last_sent_block_id_` has been set to 100 for client1, and then client2 starts a new subscription for block 0, with this change, you will be setting the last_sent_block_id metric to 100 for client2, even though TRS has not yet streamed any blocks to client2.

Reply:
Addressed. Thanks.

Reply:
It looks like now we are not persisting last_sent_block_id metric between subscriptions? For every new subscription, TRS metric `last_sent_block_id` would always be 0, even if TRS had sent x number of blocks to a client in a previous subscription round? In that case, please remove the following last line of the commit message -

```
Additionally, whenever a new subscription gets started, the metric is
updated to reflect the last block sent in the previous subscription round.
```

Reply:
Done. Thanks.